pegasus/helpers/prize_helpers.rb

```diff
@@ -17,5 +22,11 @@ def claim_prize_code(type, email, params={})
     raise
   end
 
-  DB[:hoc_survey_prizes].where(claimant: email).first[:value]
+  prize = DB[:hoc_survey_prizes].where(claimant: email, type: type).get(:value)
```

Comment:
Good call. I updated it and the test to make sure it returns the correct previously-claimed prize.